lib/mapObjects/CGTownInstance.cpp

```diff
@@ -1232,6 +1232,16 @@ void CGTownInstance::recreateBuildingsBonuses()
 		addBonusIfBuilt(BuildingID::GRAIL, Bonus::PRIMARY_SKILL, +10, PrimarySkill::ATTACK); //grail
 		addBonusIfBuilt(BuildingID::GRAIL, Bonus::PRIMARY_SKILL, +10, PrimarySkill::DEFENSE); //grail
 	}
+	else if(this->town->getFactionName() == "Cove" && hasBuilt(BuildingID::GRAIL))
```

Comment:
`this->town->getFactionName() == "Cove"` such hardcoding should be avoided.

Reply:
Will be changed later, with general solution for all grails. There's no better solution now. Cove Grail should work because it's very popular and semi-official town.

Reply:
You could use identifier, but do not use localized name.

Reply:
Accepted.